app/src/main/kotlin/com/saltedge/authenticator/features/settings/list/SettingsListViewModel.kt

```diff
@@ -105,6 +121,10 @@ class SettingsListViewModel(
                 preferenceRepository.screenshotLockEnabled = checked
                 screenshotClickEvent.postValue(ViewModelEvent(Unit))
             }
+            R.string.settings_system_dark_mode -> {
+                preferenceRepository.darkModeEnabled = checked
+                darkModeClickEvent.postValue(ViewModelEvent(Unit))
```

Comment:
apply new dark mode only if differs from current

Reply:
System dark mode selector should switch to disabled state if was clicked custom dark mode switch in MainActivity